src/Microsoft.AspNet.Hosting/HostingEngine.cs

```diff
@@ -5,251 +5,232 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading;
+using Microsoft.AspNet.Builder;
 using Microsoft.AspNet.Hosting.Builder;
-using Microsoft.AspNet.Hosting.Internal;
 using Microsoft.AspNet.Hosting.Server;
 using Microsoft.AspNet.Hosting.Startup;
+using Microsoft.Framework.ConfigurationModel;
 using Microsoft.Framework.DependencyInjection;
-using Microsoft.Framework.Runtime;
-using Microsoft.Framework.Runtime.Infrastructure;
 
 namespace Microsoft.AspNet.Hosting
 {
-    public class HostingEngine
+    public class HostingEngine : IHostingEngine
     {
-        private const string EnvironmentKey = "ASPNET_ENV";
+        private readonly IServiceCollection _applicationServiceCollection;
+        private readonly IStartupLoader _startupLoader;
+        private readonly ApplicationLifetime _applicationLifetime;
+        private readonly IHostingEnvironment _hostingEnvironment;
 
-        private readonly IServiceProvider _fallbackServices;
-        private readonly ApplicationLifetime _appLifetime;
-        private readonly IApplicationEnvironment _applicationEnvironment;
-        private readonly HostingEnvironment _hostingEnvironment;
+        // Start/ApplicationServices block use methods
+        private bool _useDisabled;
```

Comment:
I can remove the guards, but then its up to the user to beware and know that calling Use after Start() or accessing ApplicationServices is ignored / no-op